### PR TITLE
Ac 1935

### PR DESCRIPTION
**Pull request by DanielNuttall-AD**
### Description

Fixes # .

add unit test and implement stats aggregation logic

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [x] New unit tests added to cover the changes.
- [ ] New e2e tests added to cover the changes. (If it has been decided these will not be added with the PR, a separate ticket **must** be created and linked in the ticket below before merge)
- [x] All tests passed locally.

### CODE REVIEW

File: src/modules/admin/graph/graph.dto.ts [hunk 1]
```diff
@@ -98,26 +101,44 @@ export class GraphDTO {
...
-      console.log(
-        days.find((day) => {
-          moment(day.date).isSame(taskExecution.FinishedAt, 'day'); // => true
-        }),
-      );
+      const matchingDayIndex = days.findIndex((day) => {
+        return moment(day.date).isSame(taskExecution.finishedAt, 'day'); // => true
+      });
+
+      if (taskExecution.status == 'Failed') {
```
Comment (mehranrashid) [inline src/modules/admin/graph/graph.dto.ts:108]: Do we have enums for this?

File: src/modules/admin/graph/graph.dto.ts [hunk 2]
```diff
@@ -98,26 +101,44 @@ export class GraphDTO {
...
-        days.find((day) => {
-          moment(day.date).isSame(taskExecution.date, 'day'); // => true
-        }),
-      );
+    clinical_reviews.forEach((clinicalReview: IClinicalReviewExecutionDto) => {
+      const matchingDayIndex = days.findIndex((day) => {
+        return moment(day.date).isSame(clinicalReview.date, 'day'); // => true
+      });
+
+      if (clinicalReview.status == 'approved') {
```
Comment (mehranrashid) [inline src/modules/admin/graph/graph.dto.ts:128]: And enums for this?

**Review by neildsouth — APPROVED** [review]

File: src/modules/admin/graph/graph.dto.ts [hunk 3]
```diff
@@ -14,13 +15,14 @@
  *  limitations under the License.
  */
 
-import moment from 'moment';
+import moment = require('moment');
```
Comment (rowellx68) [inline src/modules/admin/graph/graph.dto.ts:18]: could we not have done `import * as moment from 'moment'`?

Comment (DanielNuttall-AD) [reply src/modules/admin/graph/graph.dto.ts]: I meant to do this, thanks for catching it. thanks :)

File: src/modules/admin/graph/graph.dto.ts (hunk 2 above)
Comment (DanielNuttall-AD) [reply src/modules/admin/graph/graph.dto.ts]: added :)

**Review by migle-markeviciute — APPROVED** [review]

**Pull request merged**